### ISSUE TITLE
AM / PM selector doesn't trigger onchange event

### ISSUE THREAD

**Issue by faalbane** [subhamayd2/day_night_time_picker#36]
If isOnChangeValueMode = true, when selecting AM / PM without touching slider, the datetime value doesn't update. 

Can you have AM / PM selection trigger an onchange event so that the value updates? Also, is there a way to get the datetime value (potential workaround when user proceeds to another screen to fetch datetime value as AM / PM doesn't trigger yet)?

**Comment by TheGlorySaint** [comment]
Hi @faalbane


i have looked into your issue.

The Boolean `isOnChangeValueMode = true` is only available for the createInlinePicker not for showPicker. I added the boolean for the createInlinePicker at the example project and when changing the Time at the Wheel it works. What you meant without touching the Slider and only changing the AM/PM Selector, I can't reproduce because the AM/PM indicator Changes on what time you select at the Slider.

Could you please give me more detailed steps to reproduce your issue?
Did you use the example project or your own project?


There is a way to get the Datetime value for the showPicker and the createInlinePicker. You can add

```
onChangeDateTime: (DateTime dateTime) {
                    print(dateTime);
                  },
```

this will give you the date and Time. You can then also add for example `print(dateTime.weekday)` to get the weekday on what you selected.

**Comment by TheGlorySaint** [comment]
Closing this issue because author of the issue didn't provided more information

**Comment by elmdecoste** [comment]
Hello, I am experiencing a similar issue. When we specifically tap the "am" or "pm" buttons above the scroll wheel in iOS mode, the `onChange` listener is not fired with an updated time object that reflects the new "am" or "pm" status. We are using the `createInlinePicker` method with `isOnChangeValueMode` set to true

**Comment by TheGlorySaint** [comment]
Hey @elmdecoste

Thanks for using day_night_time_picker.

I have checked out the example project. When I change the "am" or "pm", and adding a print statement to the onChange Method, then I get the updated time.

Could you give me a video of your issue, so i can have a look at that, because i can't reproduce it.

**Comment by elmdecoste** [comment]
@TheGlorySaint Here's a video thats showing the problem. I also attached the code that we're using to instantiate the widget

https://user-images.githubusercontent.com/1967150/211108240-35542fb4-8ca0-479a-a7e5-54d0095f491f.MP4

```
createInlinePicker(
            themeData: Theme.of(context).copyWith(
              cardColor: Theme.of(context).scaffoldBackgroundColor,
            ),
            minuteInterval: MinuteInterval.TEN,
            displayHeader: false,
            dialogInsetPadding: EdgeInsets.zero,
            value: TimeOfDay.fromDateTime(_value),
            hideButtons: true,
            isOnChangeValueMode: true,
            iosStylePicker: true,
            hourLabel: ':',
            minuteLabel: '',
            accentColor: Theme.of(context).colorScheme.primary,
            wheelHeight: 192,
            onChange: (dateTime) {
              setState(() {
                DateTime updated = DateTime(
                  _value.year,
                  _value.month,
                  _value.day,
                  dateTime.hour,
                  dateTime.minute,
                );

                _value = updated;
              });
            },
          )
```

**Comment by elmdecoste** [comment]
I believe I've fixed it with PR #89

**Comment by himsarevenus** [comment]
when @elmdecoste will be merged ? @subhamayd2
